crates/holochain_sqlite/src/db/databases.rs

```diff
@@ -13,30 +13,36 @@ pub(super) struct Databases {
     dbs: parking_lot::RwLock<HashMap<PathBuf, Box<dyn Any + Send + Sync>>>,
 }
 
-pub(super) static DATABASE_HANDLES: Lazy<Databases> = Lazy::new(|| {
-    // This is just a convenient place that we know gets initialized
-    // both in the final binary holochain && in all relevant tests
-    //
-    // Holochain (and most binaries) are left in invalid states
-    // if a thread panic!s - switch to failing fast in that case.
-    //
-    // We tried putting `panic = "abort"` in the Cargo.toml,
-    // but somehow that breaks the wasmer / test_utils integration.
+lazy_static! {
```

Comment:
We could instead do the reverse, and make a lazy_static macro that actually uses the standard types so that the api would be compatible with loom:

[https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=94d79aa57144ed265bf254f4852a3d7e](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=94d79aa57144ed265bf254f4852a3d7e)

```rust
#[macro_export]
#[doc(hidden)]
macro_rules! __lazy_static_internal {
    // optional visibility restrictions are wrapped in `()` to allow for
    // explicitly passing otherwise implicit information about private items
    ($(#[$attr:meta])* ($($vis:tt)*) static ref $N:ident : $T:ty = $init:expr; $($t:tt)*) => {
        #[allow(missing_copy_implementations)]
        #[allow(non_camel_case_types)]
        #[allow(dead_code)]
        $(#[$attr])*
        $($vis)* struct $N {__private_field: ()}
        #[doc(hidden)]
        $($vis)* static $N: $N = $N {__private_field: ()};
        impl ::core::ops::Deref for $N {
            type Target = $T;
            // this and the two __ functions below should really also be #[track_caller]
            fn deref(&self) -> &$T {
                #[inline(always)]
                fn __static_ref_initialize() -> $T { $init }

                #[inline(always)]
                fn __stability() -> &'static $T {
                    static LAZY: std::sync::OnceLock<$T> =
                        std::sync::OnceLock::new();
                    LAZY.get_or_init(|| $init)
                }
                __stability()
            }
        }
        $crate::lazy_static!($($t)*);
    };
    () => ()
}

#[macro_export]
macro_rules! lazy_static {
    ($(#[$attr:meta])* static ref $N:ident : $T:ty = $e:expr; $($t:tt)*) => {
        // use `()` to explicitly forward the information about private items
        $crate::__lazy_static_internal!($(#[$attr])* () static ref $N : $T = $e; $($t)*);
    };
    ($(#[$attr:meta])* pub static ref $N:ident : $T:ty = $e:expr; $($t:tt)*) => {
        $crate::__lazy_static_internal!($(#[$attr])* (pub) static ref $N : $T = $e; $($t)*);
    };
    ($(#[$attr:meta])* pub ($($vis:tt)+) static ref $N:ident : $T:ty = $e:expr; $($t:tt)*) => {
        $crate::__lazy_static_internal!($(#[$attr])* (pub ($($vis)+)) static ref $N : $T = $e; $($t)*);
    };
    () => ()
}

lazy_static! {
    static ref TEST_U32: u32 = 42;
    static ref TEST_STRING: String = "test".to_string();
    static ref TEST_MUTEX_STRING: std::sync::Mutex<String> =
        std::sync::Mutex::new("test1".to_string());
}

fn main() {
    println!("{}", *TEST_U32);
    println!("{}", *TEST_STRING);
    *TEST_MUTEX_STRING.lock().unwrap() = "test2".to_string();
    println!("{}", *TEST_MUTEX_STRING.lock().unwrap());
}
```

Reply:
Loom provides a type for lazy_static but not once_cell. This is one of several things I'm not very happy with in Loom.

I will take a look at this suggestion and see if I can at least leave the production code on once_cell